### PR TITLE
EIP-8

### LINKED ISSUE #349: EIP-8 (RLPx changes for homestead)
https://github.com/ethereum/EIPs/pull/49
https://github.com/ethereum/libweb3core/pull/46

### PR DESCRIPTION
**Pull request by arkpar**
cc @fjl
closes #349

### CODE REVIEW

**Comment by gavofyork** [comment]
code looks fine; is it tested it with both types of clients?

**Comment by fjl** [comment]
The EIP-8 changes to discovery are not easy to test unless you‎ temporarily change the code to add some padding and additional elements to packets.
‎
To test the new handshake, you need to implement sending of EIP-8 style auth.
I didn't implement that in libweb3core and pydevp2p yet (will do it after Homestead). You can enable sending of EIP-8 auth in geth by setting RLPX_EIP8=yes as an environment variable.
‎
I didn't look deeply, but please make sure that parsing of the devp2p hello message works according to EIP-8 requirements (ignore version mismatch, additional list elements).
‎

File: util/src/network/handshake.rs [hunk 1]
```diff
@@ -190,48 +210,105 @@ impl Handshake {
...
+		self.remote_ephemeral = try!(ec::recover(&signature, &(&shared ^ &self.remote_nonce)));
+		Ok(())
+	}
+
 	/// Parse, validate and confirm auth message
-	fn read_auth(&mut self, host: &HostInfo, data: &[u8]) -> Result<(), UtilError> {
-		trace!(target:"net", "Received handshake auth to {:?}", self.connection.socket.peer_addr());
-		if data.len() != AUTH_PACKET_SIZE {
+	fn read_auth(&mut self, secret: &Secret, data: &[u8]) -> Result<(), UtilError> {
+		trace!(target:"net", "Received handshake auth from {:?}", self.connection.socket.peer_addr());
```
Comment (gavofyork) [inline util/src/network/handshake.rs]: style for typical use of `:` would have a space after...

**Comment by arkpar** [comment]
Tested against geth's RLPX_EIP8

**Pull request merged**